src/gpu/nvidia/cudnn_convolution.cpp

```diff
@@ -31,49 +32,34 @@ status_t cudnn_convolution_fwd_t::execute_convolution(
             = utils::downcast<nvidia::sycl_cuda_stream_t *>(ctx.stream());
 
     return cuda_stream->interop_task([&](::sycl::handler &cgh) {
-        using scratch_acc_t = ::sycl::accessor<uint8_t, 1,
-                ::sycl::access::mode::read_write>;
-        auto x_acc = CTX_IN_ACCESSOR(DNNL_ARG_SRC);
-        auto weights_acc = CTX_IN_ACCESSOR(DNNL_ARG_WEIGHTS);
-        auto y_acc = CTX_OUT_ACCESSOR(DNNL_ARG_DST);
-        std::shared_ptr<
-                ::sycl::accessor<uint8_t, 1, ::sycl::access::mode::read>>
-                bias_acc;
-        std::shared_ptr<scratch_acc_t> scratch_acc;
-        std::shared_ptr<scratch_acc_t> filter_scratch_acc;
-        std::shared_ptr<scratch_acc_t> temp_dst_acc;
-        std::shared_ptr<scratch_acc_t> temp_reorder_acc;
-
-        const bool use_temp_dst = pd()->use_temp_dst();
+        auto arg_src = CTX_IN_SYCL_MEMORY(DNNL_ARG_SRC);
+        auto arg_weights = CTX_IN_SYCL_MEMORY(DNNL_ARG_WEIGHTS);
+        auto arg_dst = CTX_OUT_SYCL_MEMORY(DNNL_ARG_DST);
+        auto arg_bias = CTX_IN_SYCL_MEMORY(DNNL_ARG_BIAS);
+
+        sycl_memory_arg_t<::sycl::access::mode::read_write> arg_scratch;
+        sycl_memory_arg_t<::sycl::access::mode::read_write> arg_filter_scratch;
 
         if (with_scratchpad) {
-            scratch_acc = std::make_shared<scratch_acc_t>(
-                    utils::downcast<sycl::sycl_buffer_memory_storage_t *>(
-                            ctx.get_scratchpad_grantor()
-                                    .get_memory_storage(memory_tracking::names::
-                                                    key_conv_cudnn_algo)
-                                    .get())
-                            ->buffer()
-                            .get_access<::sycl::access::mode::read_write>(cgh));
-        }
-        if (with_bias) {
-            bias_acc = std::make_shared<
-                    ::sycl::accessor<uint8_t, 1, ::sycl::access::mode::read>>(
-                    CTX_IN_ACCESSOR(DNNL_ARG_BIAS));
+            arg_scratch = CTX_SCRATCH_SYCL_MEMORY(
```

Comment:
Can you please consider making this unconditionally one [this tweak](https://github.com/oneapi-src/oneDNN/pull/1263/files#r810377333) is implemented?

Reply:
Done.